java/vector/src/main/codegen/templates/ArrowType.java

```diff
@@ -165,7 +165,20 @@ public final T visit(${type.name?remove_ending("_")} type) {
     ${fieldType} ${field.name};
     </#list>
 
+
+    <#if type.name == "Decimal">
+    // Needed to support golden file integration tests.
+    @JsonCreator
+    public static Decimal createDecimal128(
```

Comment:
If we allow bit width other than 128, then the method name should be createDecimal?

Reply:
yes, you are right, thanks for your attention to detail.